### PR TITLE
Add any as special output format

### PR DESCRIPTION
**Pull request by Forsinge**
Resolves: #128 

TODO: 
- [x] update existing comments and comment some of the new code
- [x] test collision behaviour

To try it out you can use the `[include]` module, which I've changed to use "any" as output format. 

Collisions seem to work as intended. I tested this by adding a `[hello] -> any` transform and then importing hello from the catalog.

### CODE REVIEW

**Review by axoxelol1 — APPROVED** [review]
Seems to work as expected, nice work! Cool to see a custom deserializer as well. Will have a look at #233 once this is merged.

**Comment by CMDJojo** [comment]
I have also looked a bit at the code, it looks great! However, I do have one question; why would we need a custom any `OutputFormat`? Is it just to detect the presence of "any" in the manifest? Since it doesn't really get stored in the map, since the "any" is encoded as another variant, right? Do you think there is a good reason to have "any" as a special variant rather than checking if the OutputFormat string is "any"?

**Comment by CMDJojo** [comment]
Also, about the collision behavior; what happens if a package declares one "normal" output format and "any", like `to: [html, any]`? Do we have a custom error message for that or is it just handled like a normal collision? (Nothing I think is very important, just out of curiosity). I am unable to test the code at this moment, if you want a review from me I can do that tmrw but feel free to merge since Axel already approved!

**Comment by Forsinge** [comment]
Thanks for the review and comments. 

The custom deserializer is definitely not a necessity. As the code is structured right now we could simply perform the string comparison in `insert_transforms`. My thinking was that it is better practice to contain the string manipulations and comparisons to `OutputFormat`, and that it is less error prone to do so if we change the code later on.

I actually forgot to consider the collision scenarion in your second comment. I have yet to test it but it should result in the same error as collisions between packages. Now I'm wondering though, should we let "any" take precedence in this case, set that as `OutputFormat` for that package and ignore other specified formats?

**Comment by axoxelol1** [comment]
Hm the collision is probably a nonproblem but we should still define the expected behavior and test it. I don't think we should let `any` take precedence over other output formats if multiple are specified for a transform. If a package specifies `any` as one of multiple output formats it should probably throw an error since it does not make sense.

**Comment by CMDJojo** [comment]
Thank you for your answers!

Your implementation seems sound, and you are probably right to add an `any` `OutputFormat`. Keep it as-is!

I do not think `any` should take precedence, if any package specifies both `HTML` and `any`, there is probably an error in the module (it is simply just not intended to be done that way). If you want, you can add a check to `Package::verify` that checks for such errors, and in that way it will give an error as soon as the package is loaded rather than when the transform is loaded (it would be possible to load an erroneous package containing `any` and `html` and hide that transform, thus not yielding any error). I think that would be preferable, but it isn't a must.

But good job with the implementation!

File: core/src/lib.rs
Comment (CMDJojo) [inline core/src/lib.rs:159]: This should probably be an case-insensitive check

File: core/src/context.rs
Comment (CMDJojo) [inline core/src/context.rs:111]: These cases can be collapsed to one;
```rs
            TransformVariant::External(map) => map.get(format),
            TransformVariant::ExternalAny(t) | TransformVariant::Native(t) => Some(t),
```

**Review by CMDJojo — COMMENTED** [review]
Great job! I have only looked through the code briefly, I don't think I need to approve it since Axel already did! I found two nit-picks, but feel free to skip them if you want

**Pull request merged**
